component/http/middleware.go

```diff
@@ -123,6 +125,60 @@ func NewLoggingTracingMiddleware(path string) MiddlewareFunc {
 	}
 }
 
+type gzipResponseWriter struct {
+	io.Writer
+	http.ResponseWriter
+}
+
+// GZIPConfiguration defines configuration for GZIP Middleware.
+type GZIPConfiguration struct {
+	IgnoreRoutes []string
+}
+
+// Ignore checks if the given url ignored from compression or not.
+func (gc GZIPConfiguration) Ignore(url string) bool {
+	for _, iURL := range gc.IgnoreRoutes {
+		if strings.HasPrefix(url, iURL) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// Write provides write func to the writer.
+func (w gzipResponseWriter) Write(b []byte) (int, error) {
+	return w.Writer.Write(b)
+}
+
+// NewGzipMiddleware creates a Gzip compression middleware.
+func NewGzipMiddleware(config GZIPConfiguration) MiddlewareFunc {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if !strings.Contains(r.Header.Get("Accept-Encoding"), "gzip") || config.Ignore(r.URL.String()) {
+				next.ServeHTTP(w, r)
+				log.Debugf("url %s skipped from gzip encoding", r.URL.String())
+				return
+			}
+			// explicitly specify encoding in header
+			w.Header().Set("Content-Encoding", "gzip")
+
+			// keep content type intact
+			respHeader := r.Header.Get("Content-Type")
+			if respHeader != "" {
+				w.Header().Set("Content-Type", respHeader)
+			}
+
+			gz := gzip.NewWriter(w)
```

Comment:
Super minor nit. I'm usually a fan of small, self-contained implementations like this one, but would we want to think about supporting other compression methods as well?

 The [HTTP/1.1 RFC](https://tools.ietf.org/html/rfc2616#section-3.5) defines three possible compression methods; *gzip*, *deflate* and *compress* (based on lzw). The standard library supports all three under [compress](https://golang.org/pkg/compress/); this could be configurable by the user. 

Just talking out loud here, If it expands the scope too much, I'm not a fan either.

Reply:
I believe it is a good idea to support all three and since std Go has support for it even better...